persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/jdbc/SlickProvider.scala

```diff
@@ -179,9 +179,18 @@ private[lagom] class SlickProvider(
         }
       } else DBIO.successful(Some(schema))
     }
-
   }
 
+  // Some older JDBC drivers don't implement Connection.getSchema
+  // (including some builds of H2). This causes them to throw an
+  // AbstractMethodError at runtime.
+  // Because Try$.apply only catches NonFatal errors, and AbstractMethodError
+  // is considered fatal, we need to construct the Try explicitly.
+  private def tryGetSchema(connection: Connection): Try[String] =
```

Comment:
Lagom requires Java 8 and that included JDBC4.2 which introduced `getSchema` (http://download.oracle.com/otn-pub/jcp/jdbc-4_2-mrel2-spec/jdbc4.2-fr-spec.pdf?AuthParam=1488525841_f31869bc90688c2c9f96976b58299bef). I guess we have to require jdbcPersistence users to provide a 4.2 compatible driver.